package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -1016,6 +1016,12 @@ dnsmasq_start()
 	[ -n "$leasefile" ] && [ ! -e "$leasefile" ] && touch "$leasefile"
 	config_get_bool cachelocal "$cfg" cachelocal 1
 
+	config_get_bool logqueries "$cfg" logqueries 0
+	if [ "$logqueries" != "0" ]; then
+		config_get logfacility "$cfg" logfacility
+		[ -n "$logfacility" ] && [ ! -e "$logfacility" ] && touch "$logfacility"
```

Comment:
This logic must only execute if `logfacility` contains at least one slash, otherwise it is treated as RFC5424 syslog facility name and not as destination log file path.

Reply:
I have updated the commit to reflect this and added your comment to the commit message(hope you don't mind). Let me know if there are any issues.